### PR TITLE
First proposal for #5550 multi list ui

### PR DESCRIPTION
**Pull request by pstorch**
To better discuss #5550, here is a first proposal.

- implemented MultiChoice in ListSelection dialog

### CODE REVIEW

File: main/src/cgeo/geocaching/list/StoredList.java [hunk 1]
```diff
@@ -13,12 +13,14 @@
 import android.app.AlertDialog;
 import android.content.DialogInterface;
 import android.content.res.Resources;
+import android.widget.ListView;
```
Comment (kumy) [inline main/src/cgeo/geocaching/list/StoredList.java]: Codacy Issue found: [Avoid unused imports such as 'android.widget.ListView'](https://www.codacy.com/app/cgeo/cgeo/file/2383930731/issues/source?bid=3217125&fileBranchId=3227134#l16)

File: main/src/cgeo/geocaching/CachePopupFragment.java [hunk 2]
```diff
@@ -31,11 +38,6 @@
 import java.util.HashSet;
```
Comment (kumy) [inline main/src/cgeo/geocaching/CachePopupFragment.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.HashSet'](https://www.codacy.com/app/cgeo/cgeo/file/2383930485/issues/source?bid=3217125&fileBranchId=3227134#l38)

File: main/src/cgeo/geocaching/CachePopupFragment.java [hunk 3]
```diff
@@ -1,19 +1,26 @@
 package cgeo.geocaching;
 
+import butterknife.ButterKnife;
+
 import cgeo.geocaching.activity.Progress;
 import cgeo.geocaching.apps.navi.NavigationAppFactory;
 import cgeo.geocaching.compatibility.Compatibility;
 import cgeo.geocaching.list.StoredList;
+import cgeo.geocaching.location.Geopoint;
```
Comment (kumy) [inline main/src/cgeo/geocaching/CachePopupFragment.java]: Codacy Issue found: [Avoid unused imports such as 'cgeo.geocaching.location.Geopoint'](https://www.codacy.com/app/cgeo/cgeo/file/2383930485/issues/source?bid=3217125&fileBranchId=3227134#l9)

**Comment by pstorch** [comment]
I would like to have some feedback on this. Merge and see what NB users think about it? 
Then there is the PR from haarspalter about the move to list feature. I'm not sure what to do. Combine both or does one replace the other?

**Comment by haarspalter** [comment]
just my two cents... From my personal experience, it is sometimes hard to foresee how a new feature will be used, so I think having a MultiChoice is a good idea. Still I would prefer to offer both options to the users. Having the long press will not have influence on UI, but offer a real quite shortcut to move a cache to a certain list (still it might be the case, that I am the only one who uses the lists like this).
I did the "exercise" to move caches from 2-3 lists to another one quite some times during the weekend

**Comment by pstorch** [comment]
My personal lists are mainly static. I don't move caches around a lot. But I can understand your use case. So from my point of view, we can merge both PRs and see how it turns out. 
I changed the "Store in list" button to a "Manage" button. But I guess we can still keep the onLongPress() with the move functionality.

Any other opinions?

**Comment by rsudev** [comment]
Before offering a final opinion I would like to try your new ui a little over the next days and consider the long-press proposal in the light of its usability (I occasionally move caches around, either collecting them on some temp list for a trip or moving them to the 'found' list).
@haarspalter You are right, the long press does not affect the ui, so it will basically be 'your private feature', as it is practically impossible to discover by oneself. Regardless I would suggest that you check if really need two seemingly identical instances of ```storeCache```?

**Comment by haarspalter** [comment]
Sure, I can check to combine the two storeCache methods to a single one. Just wanted to touch as less existing code as possible in my first PR.
Managing lists can be done both in cache details, and in the cache popup. Would it make sense to check if common code could be used at those two locations? I could even have a look at this, but I guess this is done some work for another pull request, not the one that is just about the long click.

**Comment by rsudev** [comment]
In general I like the approach (the wording can be re-thought), but there are issues in the layout:

device-2016-04-19-221134

**Comment by pstorch** [comment]
Oh, we Germans are too verbose :)
Since some time I use my phone only in english, so haven't tested with it.
I'll have a look. 
Having all three buttons in one line is too much. Maybe I try to put the "Manage" button in the next line with the lists.

**Comment by Bananeweizen** [comment]
I'm going to merge this, cleaning up the text clashes should be easier afterwards.

**Comment by Bananeweizen** [comment]
BTW: Android lint throws a warning specifically for the above overlapping text: http://ci.cgeo.org/job/cgeo%20continuous%20integration/568/androidLintResult/new/package.-1249878641/file.-148985145/

**Comment by pstorch** [comment]
Interesting, when I run the lint checker locally in AS, I also get a warning at line 96:
`LinearLayout-3 can overlap @id/offline_text if LinearLayout-3 grows due to localized text expansion (at line 96)`
First time I see this kind of issues, so I have to think about it.

**Comment by Bananeweizen** [comment]
I've started setting the tools:text="some useful text" and tools:visibility="visible" attributes in a lot of our layouts. Feel free to add more dummy text if you want to test the overlap issues already in the layout editor without even running the app. E.g. take this commit as an example: 449baf6. If you now pull and have a look at the cache details page in the layout editor again, you will see that everything flows correctly even with very long strings.

**Pull request merged**